### PR TITLE
refactor to implement the assignment folder structure

### PR DESCRIPTION
**Pull request by Kemal-Wuzhi**

### CODE REVIEW

**Review by lochuhsin — COMMENTED** [review]
`python

CreateUser, GetUser, PutUser, DeleteUser

File: service/user_service.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import smtplib
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+
+def send_email(to_email, subject, body):
+    EMAIL_HOST = 'example.com'
+    EMAIL_PORT = 465  
+    EMAIL_HOST_USER = 'kemal@mentalhealth.com'
+    EMAIL_HOST_PASSWORD = '12345678'
```
Comment (lochuhsin) [inline service/user_service.py:9]: environ variable
```python
import os

env = os.environ -> dictionary
```

File: env/local.env [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+POSTGRES_PASSWORD=admin
+POSTGRES_USER=admin
+POSTGRES_DB=backend
+POSTGRES_HOST=backend_db
+POSTGRES_PORT=5432
+USE_PGBOUNCER=False
+PGBOUNCER_HOST=""
+
+# Redis
+REDIS_HOST=redis
```
Comment (lochuhsin) [inline env/local.env:11]: #### SMTP SERVER SETTING
SMTP_EMAIL = 
SMTP_PASSWORD =

File: service/user_service.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    msg = MIMEMultipart()
+    msg['From'] = EMAIL_HOST_USER
+    msg['To'] = to_email
+    msg['Subject'] = subject
+    msg.attach(MIMEText(body, 'plain'))
+
```
Comment (lochuhsin) [inline service/user_service.py:16]:
```python
from enum import Enum

class UserType(str, Enum):
      Patient
      Doctor
      Admin

def smtp_instance_selector(position: UserType):
       if position == UserType.Patinent:
             return get_patient_smtp_instance()

Comment (lochuhsin) [reply service/user_service.py:16]:
```python
if usertype == "patient":
    ...
    
elif usertype == "docter":
    ...

File: service/user_service.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    try:
+        server = smtplib.SMTP_SSL(EMAIL_HOST, EMAIL_PORT)  
+        server.login(EMAIL_HOST_USER, EMAIL_HOST_PASSWORD)
+        server.sendmail(EMAIL_HOST_USER, to_email, msg.as_string())
+    except Exception as e:
+        print(f"Error sending email: {e}")
+    finally:
+        server.quit()
```
Comment (lochuhsin) [inline service/user_service.py:24]: Use singleton pattern

Design Pattern -> Singleton
create a singleton.py
initialize smtp server with email/password/port ...etc

```python3


from singleton import smtp_instance

smtp_instance.send_email(........)

File: service/user_service.py (hunk 1 above)
Comment (Kemal-Wuzhi) [reply service/user_service.py]: fixed it !

File: service/user_service.py (hunk 3 above)
Comment (Kemal-Wuzhi) [reply service/user_service.py:66]: fixed it

File: service/user_service.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    cred = None
+    json_file = f'token_{api_service_name}_{api_version}.json'
+    # json_data = json.dumps(json_file)
+    
+
+    if os.path.exists(json_file):
+        print("EXISTS")
+        with open(json_file, 'rb') as token:
+            print("opened")
+            cred2=json.loads(token)
```
Comment (lochuhsin) [inline service/user_service.py:56]: json.load

**Pull request merged**